win32ss/gdi/eng/gradient.c

```diff
@@ -251,7 +254,7 @@ IntEngGradientFillRect(
   FINITCOL(linefrom, lineto, 0); \
   FINITCOL(linefrom, lineto, 1); \
   FINITCOL(linefrom, lineto, 2); \
-  for(g = sx[linefrom]; g != sx[lineto]; g += gxi) \
+  for(g = sx[linefrom] - 1; g != sx[lineto] + gxi; g += gxi) \
```

Comment:
Could you try the following?
```suggestion
  for(g = sx[linefrom] - 1; g <= sx[lineto]; g += gxi) \
```